### PR TITLE
Blog-app-with-Rails: Set up Rspec tests on the controllers

### PR DESCRIPTION
**Pull request by Sonickmumba**
In this milestone, I

- Add RSpec as a gem to your project.
- Create a Request spec file for all your controllers.
- Make sure that for each action you check:
    - If response status was correct.
    - If a correct template was rendered.
    - If the response body includes correct placeholder text.

Screen Shot 2022-12-20 at 1 27 26 AM

### CODE REVIEW

File: spec/controllers/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    end
+  end
+
+  describe 'posts#show' do
+    it 'render the show template' do
+      get '/users/2/posts/2'
+      expect(response).to render_template('show')
+    end
+  end
+end
```
Comment (ShadyShawkat) [inline spec/controllers/posts_controller_spec.rb]: - [ ] Kindly implement all of the following tests for each of the actions in each controller:
   - If the response status was correct.
   - If a correct template was rendered.
   - If the response body includes correct placeholder text.
- As per [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=Make%20sure%20that%20for,includes%20correct%20placeholder%20text.) requirement.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @Sonickmumba, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/controllers/posts_controller_spec.rb (hunk 1 above)
Comment (Sonickmumba) [reply spec/controllers/posts_controller_spec.rb]: @ShadyShawkat thanks for the review. Kindly note that all the three requirements have been implemented. Please go through again. Thanks.

**Review by Reem-lab — APPROVED** [review]
# Hi @Sonickmumba, 👋🏻👋🏻 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉🚀🔥🎉🎉 


## Highlights ✨✨
- Followed Gitflow ✅
- Descriptivr PR ✅
- Great working on the testing🔥🚀🎉
- Professional README ✅


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**